import "SettingMessage.proto";
import "InitClient.proto";
import "ChunkData.proto";
import "ChunkZone.proto";
import "Displacement.proto";
import "PingTime.proto";

message ProtocolMessage {
  enum PacketContent
  {
    ANSWER = 0;
    SETTING = 1;
    CLINIT = 2;
    CHUNK = 3;
    CHUNKZONE = 4;
    DISPLACEMENT = 5;
  }
  required PacketContent content = 1;
  optional SettingMessage settings = 2;
  optional InitClient clInit = 3;
  optional ChunkData chunkData = 4;
  optional WorldZone worldZone = 5;
  optional Displacement displacement = 6;
}
